js/compressed-token/src/program.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  PublicKey,
+  Keypair,
+  Connection,
+  TransactionInstruction,
+  SystemProgram,
+} from '@solana/web3.js';
+import { BN, Program, AnchorProvider, setProvider } from '@coral-xyz/anchor';
+import { IDL, PspCompressedToken } from './idl/psp_compressed_token';
+import {
+  LightSystemProgram,
+  UtxoWithMerkleContext,
+  UtxoWithMerkleProof,
+  bn,
+  confirmConfig,
+  defaultStaticAccountsStruct,
+  toArray,
+  useWallet,
+} from '@lightprotocol/stateless.js';
+import {
+  MINT_SIZE,
+  TOKEN_PROGRAM_ID,
+  createInitializeMint2Instruction,
+} from '@solana/spl-token';
+import { MINT_AUTHORITY_SEED, POOL_SEED } from './constants';
+import { Buffer } from 'buffer';
+
+export type CompressedTransferParams = {
+  /** Utxos with lamports to spend as transaction inputs */
+  fromBalance: // TODO: selection upfront
+  | UtxoWithMerkleContext
+    | UtxoWithMerkleProof
+    | (UtxoWithMerkleContext | UtxoWithMerkleProof)[];
+  /** Solana Account that will receive transferred compressed lamports as utxo  */
+  toPubkey: PublicKey;
+  /** Amount of compressed lamports to transfer */
+  amount: number | BN;
+  /** Mint public key */
+  mint: PublicKey;
+  // /** TODO: Optional: if different feepayer than owner of utxos */
+  // payer?: PublicKey;
+};
+
+/** Create Mint account for compressed Tokens */
+export type CreateMintParams = {
+  /** Tx feepayer */
+  feePayer: PublicKey;
+  /** Mint authority */
+  authority: PublicKey;
+  /** Mint public key */
+  mint: PublicKey;
+  /** Mint decimals */
+  decimals: number;
+  /** Optional: freeze authority */
+  freezeAuthority: PublicKey | null;
+  /** lamport amount for mint account rent exemption */
+  rentExemptBalance: number;
+};
+
+/**
+ * Create compressed token accounts
+ */
+export type MintToParams = {
+  /** Tx feepayer */
+  feePayer: PublicKey;
+  /** Mint authority */
+  authority: PublicKey;
+  /** Mint public key */
+  mint: PublicKey;
+  /** The Solana Public Keys to mint to. Accepts batches */
+  toPubkey: PublicKey[] | PublicKey;
+  /** The amount of compressed tokens to mint. Accepts batches */
+  amount: BN | BN[] | number | number[]; // TODO: check if considers mint decimals
+  /** Public key of the state tree to mint into. */
+  merkleTree: PublicKey; // TODO: make optional with default system state trees
+};
+
+export class CompressedTokenProgram {
+  /**
+   * @internal
+   */
+  constructor() {}
+
+  /**
+   * Public key that identifies the CompressedPda program
+   */
+  static programId: PublicKey = new PublicKey(
+    // TODO: can add check to ensure its consistent with the idl
+    '9sixVEthz2kMSKfeApZXHwuboT6DZuT6crAYJTciUCqE',
```

Comment:
if we have it in constants as well probably makes sense to get if from there.

Reply:
for CompressedPdaProgram and CompressedTokenProgram we dont have it in constants since we have the program.ts classes, for these IMO best to just define it here _or_ get it directly from the IDL 

i.e. all consumers will just call `CompressedTokenProgram.programId` to get the value